dcpy/test/lifecycle/distribute/test_bytes.py

```diff
@@ -0,0 +1,132 @@
+)
+DATA_DICT_NAME_IN_ZIP_FILE = "zipped_data_dict.pdf"
+
+ZIP_FILE = ds.Package(
+    id="my_zip",
+    filename="my.zip",
+    contents=[
+        ds.PackageFile(id=SHAPEFILE.id),
+        ds.PackageFile(id=DATA_DICT.id, filename=DATA_DICT_NAME_IN_ZIP_FILE),
+    ],
+)
+
+BYTES_DEST_WITH_INDIVIDUAL_FILES = "bytes_dest_with_individual_files"
+BYTES_DEST_WITH_ZIP = "bytes_dest_with_zip"
+
+SHP_URL = "https://s-media.nyc.gov/agencies/dcp/assets/files/my_shp.zip"
+DATA_DICT_URL = "https://s-media.nyc.gov/agencies/dcp/assets/files/my_data_dict.pdf"
+ZIP_URL = "https://s-media.nyc.gov/agencies/dcp/assets/files/zip.pdf"
+
+SOCRATA_DEST_ID = "soc_dest"
+
+
+def make_metadata():
+    return ds.Metadata(
+        id="test",
+        attributes=ds.DatasetAttributes(
+            display_name="attrs_display_name",
+            description="attrs_description",
+            each_row_is_a="attrs_each_row_is_a",
+            tags=["attrs_1", "attrs_2"],
+            custom={
+                "custom_attr_key": "custom_attr_val",
+                "custom_attr_key_to_override": "custom_attr_val_to_override",
+            },
+        ),
+        columns=[],
+        files=[
+            ds.FileAndOverrides(
+                file=SHAPEFILE,
+            ),
+            ds.FileAndOverrides(
+                file=DATA_DICT,
+            ),
+        ],
+        destinations=[
+            ds.DestinationWithFiles(
+                id=BYTES_DEST_WITH_INDIVIDUAL_FILES,
+                type="bytes",
+                files=[
+                    ds.DestinationFile(
+                        id=SHAPEFILE.id,
+                        custom={"url": SHP_URL},
+                    ),
+                    ds.DestinationFile(
+                        id=DATA_DICT.id,
+                        custom={"url": DATA_DICT_URL},
+                    ),
+                    ds.DestinationFile(
+                        id=ZIP_FILE.id,
+                        custom={"url": ZIP_URL},
+                    ),
+                ],
+            ),
+            ds.DestinationWithFiles(id=SOCRATA_DEST_ID, type="socrata", files=[]),
+        ],
+        assembly=[ZIP_FILE],
+    )
+
+
+def test_plan():
+    assert {
+        SHAPEFILE.id: {"path": f"dataset_files/{SHAPEFILE.filename}", "url": SHP_URL},
+        DATA_DICT.id: {
+            "path": f"attachments/{DATA_DICT.filename}",
+            "url": DATA_DICT_URL,
+        },
+        ZIP_FILE.id: {"path": f"zip_files/{ZIP_FILE.filename}", "url": ZIP_URL},
+    } == bytes._get_file_url_mappings_by_id(
+        make_metadata(), BYTES_DEST_WITH_INDIVIDUAL_FILES
+    )
+
+
+def test_plan_errors_for_socrata():
+    with pytest.raises(Exception, match=bytes.NON_BYTES_DEST_ERROR):
+        plan = bytes.pull_destination_files(Path(""), make_metadata(), SOCRATA_DEST_ID)
+
+
+@patch("dcpy.lifecycle.package.assemble.unpackage")
+@patch("urllib.request.urlretrieve")
+def test_pull_destination_files_mocked(mock_urlretrieve, mock_unpackage, tmp_path):
```

Comment:
this test failed with
```bash
AttributeError: <module 'dcpy.lifecycle.package.assemble' from
'/__w/data-engineering/data-engineering/dcpy/lifecycle/package/assemble.py'> does not have the attribute 'unpackage'
```

maybe line 105 should be `@patch("dcpy.lifecycle.package.assemble.unzip_into_package")` ?

Reply:
yeah, had just fixed that! Not sure how let that one slip through.